src/graph/sampler.cc

```diff
@@ -1227,13 +1227,7 @@ NegSubgraph EdgeSamplerObject::genNegEdgeSubgraph(const Subgraph &pos_subg,
   neg_subg.graph = GraphPtr(new ImmutableGraph(neg_coo));
   neg_subg.induced_vertices = induced_neg_vid;
   neg_subg.induced_edges = induced_neg_eid;
-  // If we didn't sample all nodes to form negative edges, some of the nodes
-  // in the vector might be redundant.
-  if (neg_sample_size < num_tot_nodes) {
-    std::sort(neg_vids.begin(), neg_vids.end());
-    auto it = std::unique(neg_vids.begin(), neg_vids.end());
-    neg_vids.resize(it - neg_vids.begin());
-  }
```

Comment:
should we simply remove it? the subgraph will have duplicated nodes if we just remove it.

Reply:
It seems OK for subgraph to have duplicated nodes. If we remove the duplicated nodes, we should sample more nodes to make the num_of_edges aligned. 
I think we need another PR to cleanup our neg-sampling code.

Reply:
agreed.

Reply:
we should remove this change from this PR. We should separate DGL-ke modifications from DGL core modifications. we might need to create a minor version release of DGL core for DGL app packages. To create a minor version release, it's usually cherry-pick commits from the master branch. Having the modification in this PR makes it difficult to create minor release of DGL.

Reply:
This modification is necessary. The sampler semantic has changed slightly.